src/species_advance/species_advance.cc

```diff
@@ -35,7 +35,30 @@ checkpt_species( const species_t * sp )
                   sp->max_np * sizeof(size_t), 1, 1, 128 );
   }
   #endif
+  #ifdef VPIC_PARTICLE_ANNOTATION
+  if(sp->has_annotation) {
+    checkpt_data( sp->p_annotation,
+                  sp->np     * sp->has_annotation * sizeof(float),
+                  sp->max_np * sp->has_annotation * sizeof(float), 1, 1, 128);
+  }
+  #endif
 
+  checkpt_data(sp->output_buffer_dx, sp->buf_size*sizeof(float),   sp->buf_size*sizeof(float),   1, 1, 128);
+  checkpt_data(sp->output_buffer_dy, sp->buf_size*sizeof(float),   sp->buf_size*sizeof(float),   1, 1, 128);
+  checkpt_data(sp->output_buffer_dz, sp->buf_size*sizeof(float),   sp->buf_size*sizeof(float),   1, 1, 128);
+  checkpt_data(sp->output_buffer_i,  sp->buf_size*sizeof(int64_t), sp->buf_size*sizeof(int64_t), 1, 1, 128);
+  checkpt_data(sp->output_buffer_ux, sp->buf_size*sizeof(float),   sp->buf_size*sizeof(float),   1, 1, 128);
+  checkpt_data(sp->output_buffer_uy, sp->buf_size*sizeof(float),   sp->buf_size*sizeof(float),   1, 1, 128);
+  checkpt_data(sp->output_buffer_uz, sp->buf_size*sizeof(float),   sp->buf_size*sizeof(float),   1, 1, 128);
+  checkpt_data(sp->output_buffer_w,  sp->buf_size*sizeof(float),   sp->buf_size*sizeof(float),   1, 1, 128);
+  #ifdef VPIC_GLOBAL_PARTICLE_ID
+  checkpt_data(sp->output_buffer_id, sp->buf_size*sizeof(size_t),  sp->buf_size*sizeof(size_t),  1, 1, 128);
+  #endif
+  #ifdef VPIC_PARTICLE_ANNOTATION
+  checkpt_data(sp->output_buffer_an, sp->buf_size*sp->buf_n_annotation*sizeof(float), sp->buf_size*sp->buf_n_annotation*sizeof(float), 1, 1, 128);
```

Comment:
This stuff might be fun to port to Kokkos..

Reply:
I feel bad dropping that into your lap as is. If only I got payed to put this into vpic-kokkos too.